ssh-pairing-server.c

```diff
@@ -59,13 +98,21 @@ int main(int argc, char *argv[])
 	// the callback-based pubkey handling is not possible:
 	// https://gitlab.com/libssh/libssh-mirror/-/issues/146
 
+	/*
+	 * I don't really get why the INTERACTIVE auth method is offered in
+	 * the first place. It looks like the only purpose is to send the
+	 * "Received %d public keys" message to the connecting client. This
+	 * should be documented better.
+	 */
 	ssh_set_auth_methods(session, SSH_AUTH_METHOD_PUBLICKEY | SSH_AUTH_METHOD_INTERACTIVE);
 
 	char *authorized_keys = strdup("");
 	int keycount = 0;
 
 	ssh_message message;
 	while ((message = ssh_message_get(session))) {
+		/* I would move the processing done in the while loop into a
```

Comment:
Then I'd either need global variables or have to pass multiple pointers...

Reply:
You could introduce a `struct` keeping the runtime program state this could generally improve readability.